GDJS/Runtime/runtimeobject.ts

```diff
@@ -1617,40 +1618,128 @@ namespace gdjs {
+                  moveXArray.push(result.move_axis[0]);
+                  moveYArray.push(result.move_axis[1]);
                 }
               }
             }
           }
         }
       }
+      return this._doSeparateFromObjects(moveXArray, moveYArray);
+    }
+
+    /**
+     * Separate the object from others objects, using the results from collisionTest call.
+     * @param moveXArray
+     * @param moveYArray
+     * @return true if the object was moved
+     */
+    private _doSeparateFromObjects(
+      moveXArray: Array<float>,
+      moveYArray: Array<float>
+    ): boolean {
+      if (moveXArray.length === 0) {
+        moveXArray.length = 0;
+        moveYArray.length = 0;
+        return false;
+      }
+      if (moveXArray.length === 1) {
+        // Move according to the results returned by the collision algorithm.
+        this.setPosition(
+          this.getX() + moveXArray[0],
+          this.getY() + moveYArray[0]
+        );
+        moveXArray.length = 0;
+        moveYArray.length = 0;
+        return true;
+      }
+
+      // Find the longest vector
+      let distanceSqMax = 0;
+      let distanceMaxIndex = 0;
+      for (let index = 0; index < moveXArray.length; index++) {
+        const moveX = moveXArray[index];
+        const moveY = moveYArray[index];
+
+        const distanceSq = moveX * moveX + moveY * moveY;
+        if (distanceSq > distanceSqMax) {
+          distanceSqMax = distanceSq;
+          distanceMaxIndex = index;
+        }
+      }
+
+      const distanceMax = Math.sqrt(distanceSqMax);
+      // unit vector of the longest vector
+      const uX = moveXArray[distanceMaxIndex] / distanceMax;
+      const uY = moveYArray[distanceMaxIndex] / distanceMax;
+
+      // normal vector of the longest vector
+      const vX = -uY;
+      const vY = uX;
+
+      // Project other vectors on the normal
+      let scalarProductMin = 0;
+      let scalarProductMax = 0;
+      for (let index = 0; index < moveXArray.length; index++) {
+        const moveX = moveXArray[index];
+        const moveY = moveYArray[index];
+
+        const scalarProduct = moveX * vX + moveY * vY;
+        scalarProductMin = Math.min(scalarProductMin, scalarProduct);
+        scalarProductMax = Math.max(scalarProductMax, scalarProduct);
+      }
 
-      //Move according to the results returned by the collision algorithm.
-      this.setPosition(this.getX() + xMove, this.getY() + yMove);
-      return moved;
+      // Apply the longest vector
+      let deltaX = moveXArray[distanceMaxIndex];
+      let deltaY = moveYArray[distanceMaxIndex];
+
+      // Apply the longest projected vector if they all are in the same direction
+      // Some projections could have rounding errors,
+      // they are considered negligible under a 1 for 1,000,000 ratio.
+      if (
+        -scalarProductMin < scalarProductMax / 1048576 !==
+        scalarProductMax < -scalarProductMin / 1048576
+      ) {
+        if (scalarProductMin !== 0) {
```

Comment:
If scalarProductMin is very very very small (-0.0001) and scalarProductMax is very large (10000), then we will be there (because 
`-scalarProductMin < scalarProductMax / 1048576 !== scalarProductMax < -scalarProductMin / 1048576` => `0.0001 < 10000 / 1048576 !== 10000 < 0.0001 / 1048576` => `0.0001 < 0.0095 !== 10000 < 9.5e-11` => `true !== false` => `true`)

BUT because scalarProductMin is not exactly 0, we will move according to scalarProductMin.

Should we instead look at if scalarProductMin is not negligible against scalarProductMax?